lib/data-access/bookmarks.ts

```
@@ -55,87 +61,89 @@ async function fetchExternalBookmarks(): Promise<UnifiedBookmark[] | null> {
-    console.log('[DataAccess/Bookmarks] Returning bookmarks from cache (skipExternalFetch=true).');
-    return cached.bookmarks;
+  const bookmarksLength = cached?.bookmarks?.length;
+  if (typeof bookmarksLength === 'number' && bookmarksLength > 0) {
+    // console.log('[Bookmarks] returning ${cached.bookmarks.length} cached bookmarks');
+    if (!skipExternalFetch && ServerCacheInstance.shouldRefreshBookmarks()) {
+      // console.log('[Bookmarks] cache stale, enqueueing background refresh');
+      enqueueBookmarkRefresh();
+    } else if (ServerCacheInstance.shouldRefreshBookmarks()) {
+      // console.log('[Bookmarks] cache stale but external fetch skipped, not enqueueing refresh');
+    }
+    return cached!.bookmarks;
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)